plugins/single_plugins/scale.cpp

```diff
@@ -595,43 +595,52 @@ class wayfire_scale : public wf::plugin_interface_t
         for (auto& e : scale_data)
         {
             auto view = e.first;
-            if (!view || !scale_data[view].transformer || !scale_view(view))
+            auto& view_data = e.second;
+            if (!view || !view_data.transformer || !scale_view(view))
             {
                 continue;
             }
 
-            scale_data[view].transformer->scale_x =
-                scale_data[view].animation.scale_animation.scale_x;
-            scale_data[view].transformer->scale_y =
-                scale_data[view].animation.scale_animation.scale_y;
-            scale_data[view].transformer->translation_x =
-                scale_data[view].animation.scale_animation.translation_x;
-            scale_data[view].transformer->translation_y =
-                scale_data[view].animation.scale_animation.translation_y;
-            scale_data[view].transformer->alpha = scale_data[view].fade_animation;
+            view_data.transformer->scale_x =
+                view_data.animation.scale_animation.scale_x;
+            view_data.transformer->scale_y =
+                view_data.animation.scale_animation.scale_y;
+            view_data.transformer->translation_x =
+                view_data.animation.scale_animation.translation_x;
+            view_data.transformer->translation_y =
+                view_data.animation.scale_animation.translation_y;
+            view_data.transformer->alpha = view_data.fade_animation;
 
             view->damage();
 
             for (auto& child : view->children)
             {
-                view = child;
-                if (!view || !scale_data[view].transformer)
+                if (!child)
                 {
                     continue;
                 }
 
-                scale_data[view].transformer->scale_x =
-                    scale_data[view].animation.scale_animation.scale_x;
-                scale_data[view].transformer->scale_y =
-                    scale_data[view].animation.scale_animation.scale_y;
-                scale_data[view].transformer->translation_x =
-                    scale_data[view].animation.scale_animation.translation_x;
-                scale_data[view].transformer->translation_y =
-                    scale_data[view].animation.scale_animation.translation_y;
-                scale_data[view].transformer->alpha =
-                    scale_data[view].fade_animation;
-
-                view->damage();
+                auto it = scale_data.find(child);
+                if ((it == scale_data.end()) || !it->second.transformer)
+                {
+                    /* this should not happen?
+                     * child views can show up here before they are attached */
```

Comment:
The two lines in this comment seem to contradict each other.

Reply:
I'm not sure. At first I thought this should not happen, but then realized this can happen under normal circumstances (a child view is created, but not visible yet?).
Anyway, I'll have to double check this in debugger and then I'll update the comment.

Reply:
Looking in a debugger, this is the sequence of events (in the case of gedit, for the "close without saving" dialog):
1. view_attached() is called
2. child view shows up here, but does not have a transformer set up, so the continue statement is hit on line 628
3. view_geometry_changed() is called which results in the transformer set up (via layout_slots())
So I think it is normal for this case to be hit